### PR TITLE
Get providers

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

### CODE REVIEW

File: src/main/java/org/gridsuite/securityanalysis/server/SecurityAnalysisController.java [hunk 1]
```diff
@@ -151,4 +152,19 @@ public ResponseEntity<Mono<Void>> stop(@Parameter(description = "Result UUID") @
         Mono<Void> result = service.stop(resultUuid, receiver);
         return ResponseEntity.ok().body(result);
     }
+
+    @GetMapping(value = "/providers", produces = APPLICATION_JSON_VALUE)
+    @Operation(summary = "Get all security analysis providers")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Security analysis providers have not been found")})
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/securityanalysis/server/SecurityAnalysisController.java:158]: Security analysis providers have been found

File: src/main/java/org/gridsuite/securityanalysis/server/SecurityAnalysisController.java [hunk 2]
```diff
@@ -151,4 +152,19 @@ public ResponseEntity<Mono<Void>> stop(@Parameter(description = "Result UUID") @
...
+    public ResponseEntity<List<String>> getProviders() {
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON)
+                .body(service.getProviders());
+    }
+
+    @GetMapping(value = "/default-provider", produces = TEXT_PLAIN_VALUE)
+    @Operation(summary = "Get security analysis default provider")
+    @ApiResponses(@ApiResponse(responseCode = "200", description = "The security analysis default provider has not been found"))
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/securityanalysis/server/SecurityAnalysisController.java:166]: The security analysis default provider has been found

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/securityanalysis/server/SecurityAnalysisController.java (hunk 1 above)
Comment (geofjamg) [reply src/main/java/org/gridsuite/securityanalysis/server/SecurityAnalysisController.java]: Done

File: src/main/java/org/gridsuite/securityanalysis/server/SecurityAnalysisController.java (hunk 2 above)
Comment (geofjamg) [reply src/main/java/org/gridsuite/securityanalysis/server/SecurityAnalysisController.java]: Done

**Review by bendaamerahm — APPROVED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**